pandas/core/generic.py

```diff
@@ -5401,40 +5401,59 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
 
     def asof(self, where, subset=None):
         """
-        The last row without any NaN is taken (or the last row without
-        NaN considering only the subset of columns in the case of a DataFrame)
-
-        .. versionadded:: 0.19.0 For DataFrame
-
-        If there is no good value, NaN is returned for a Series
-        a Series of NaN values for a DataFrame
-
+        Take the last row without any NaN. 
+        
+        Or the last row without NaN considering only the subset of columns in the case of a DataFrame.
+        If there is no matched value, NaN is returned for a Series a Series of NaN values for a DataFrame.
+        
+        .. versionadded:: 0.19.0 For DataFrame.
+        
         Parameters
         ----------
-        where : date or array of dates
-        subset : string or list of strings, default None
-           if not None use these columns for NaN propagation
+        where : datetime
+            Date or array of dates.
+        subset : str
```

Comment:
subset: str or list of strings
    if not None, use the columns for NaN propagation.